src/main/flight/rpm_filter.c

```diff
@@ -45,162 +44,119 @@
 
 #include "rpm_filter.h"
 
-#define RPM_FILTER_MAXHARMONICS 3
-#define SECONDS_PER_MINUTE      60.0f
-#define ERPM_PER_LSB            100.0f
-#define MIN_UPDATE_T            0.001f
+#define RPM_FILTER_HARMONICS_MAX 3
+#define RPM_FILTER_DURATION_S    0.001f  // Maximum duration allowed to update all RPM notches once
+#define SECONDS_PER_MINUTE       60.0f
+#define ERPM_PER_LSB             100.0f
 
 
-static pt1Filter_t rpmFilters[MAX_SUPPORTED_MOTORS];
+typedef struct rpmFilter_s {
 
-typedef struct rpmNotchFilter_s {
+    int numHarmonics;
+    float minHz;
+    float maxHz;
+    float fadeRangeHz;
+    float q;
 
-    uint8_t  harmonics;
-    float    minHz;
-    float    maxHz;
-    float    fadeRangeHz;
-    float    q;
     timeUs_t looptimeUs;
+    biquadFilter_t notch[XYZ_AXIS_COUNT][MAX_SUPPORTED_MOTORS][RPM_FILTER_HARMONICS_MAX];
 
-    biquadFilter_t notch[XYZ_AXIS_COUNT][MAX_SUPPORTED_MOTORS][RPM_FILTER_MAXHARMONICS];
+} rpmFilter_t;
 
-} rpmNotchFilter_t;
+// Singleton
+FAST_DATA_ZERO_INIT static rpmFilter_t rpmFilter;
 
-FAST_DATA_ZERO_INIT static float   erpmToHz;
-FAST_DATA_ZERO_INIT static float   filteredMotorErpm[MAX_SUPPORTED_MOTORS];
-FAST_DATA_ZERO_INIT static float   motorFrequency[MAX_SUPPORTED_MOTORS];
-FAST_DATA_ZERO_INIT static float   minMotorFrequency;
-FAST_DATA_ZERO_INIT static uint8_t numberFilters;
-FAST_DATA_ZERO_INIT static uint8_t numberRpmNotchFilters;
-FAST_DATA_ZERO_INIT static uint8_t filterUpdatesPerIteration;
-FAST_DATA_ZERO_INIT static float   pidLooptime;
-FAST_DATA_ZERO_INIT static rpmNotchFilter_t filters[2];
-FAST_DATA_ZERO_INIT static rpmNotchFilter_t *gyroFilter;
+FAST_DATA_ZERO_INIT static pt1Filter_t motorFreqLpf[MAX_SUPPORTED_MOTORS];
```

Comment:
Should we use `motorFrequencyLpf` to be inline with other variables?

Reply:
Everything is not inlined on purpose to not having to adjust the inlining of whole code blocks on future changes.

Reply:
Did mean renaming to `motorFrequencyLpf` 😂